### PR TITLE
Bug 152556: WebGL based canvases composite incorrectly after changing size

### PR DESCRIPTION
**Pull request by graouts**

### CODE REVIEW

File: Source/WebCore/platform/graphics/mac/GraphicsContext3DMac.mm [hunk 1]
```diff
@@ -342,10 +342,10 @@ static void setPixelFormat(Vector<CGLPixelFormatAttribute>& attribs, int colorBi
 #if PLATFORM(IOS)
 bool GraphicsContext3D::setRenderbufferStorageFromDrawable(GC3Dsizei width, GC3Dsizei height)
 {
-    [m_webGLLayer setBounds:CGRectMake(0, 0, width, height)];
+    [m_webGLLayer setBackingStoreSize:CGSizeMake(width, height)];
     [m_webGLLayer setOpaque:(m_internalColorFormat != GL_RGBA8)];
```
Comment (graouts) [inline Source/WebCore/platform/graphics/mac/GraphicsContext3DMac.mm:346]: I suppose this should now be set directly in the WebGLLayer initializer or in `setBackingStoreSize`.

File: Source/WebCore/platform/graphics/mac/WebGLLayer.h [hunk 2]
```diff
@@ -49,6 +52,11 @@ namespace WebCore {
 
 - (CGImageRef)copyImageSnapshotWithColorSpace:(CGColorSpaceRef)colorSpace;
 
+#if PLATFORM(IOS)
+@property (readonly, nonatomic) CAEAGLLayer* glLayer;
+- (void)setBackingStoreSize:(CGSize)size;
```
Comment (graouts) [inline Source/WebCore/platform/graphics/mac/WebGLLayer.h:57]: I was wondering if I can have both of those in the same `#if PLATFORM(IOS)` block or if there needs to be one for the property and one for the method.

File: Source/WebCore/platform/graphics/mac/WebGLLayer.mm [hunk 3]
```diff
@@ -45,12 +45,22 @@
 @implementation WebGLLayer
 
 @synthesize context=_context;
+@synthesize glLayer=_glLayer;
```
Comment (graouts) [inline Source/WebCore/platform/graphics/mac/WebGLLayer.mm:48]: I suppose this needs to be conditionalized as well.

File: Source/WebCore/platform/graphics/mac/GraphicsContext3DMac.mm (hunk 1 above)
Comment (grorg) [inline Source/WebCore/platform/graphics/mac/GraphicsContext3DMac.mm:346]: What "this" do you mean?

File: Source/WebCore/platform/graphics/mac/WebGLLayer.h (hunk 2 above)
Comment (grorg) [inline Source/WebCore/platform/graphics/mac/WebGLLayer.h:57]: I think it is fine in one block.

File: Source/WebCore/platform/graphics/mac/GraphicsContext3DMac.mm (hunk 1 above)
Comment (graouts) [inline Source/WebCore/platform/graphics/mac/GraphicsContext3DMac.mm:346]: The call to `[m_webGLLayer setOpaque:(m_internalColorFormat != GL_RGBA8)];`

**Pull request closed without merging**